src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -0,0 +1,214 @@
+#include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
+
+#include "control/controlobject.h"
+#include "controllers/controller.h"
+#include "controllers/controllerdebug.h"
+#include "controllers/scripting/colormapperjsproxy.h"
+#include "controllers/scripting/legacy/controllerscriptinterface.h"
+#include "errordialoghandler.h"
+#include "mixer/playermanager.h"
+
+ControllerScriptEngineLegacy::ControllerScriptEngineLegacy(Controller* controller)
+        : ControllerScriptEngineBase(controller) {
+}
+
+bool ControllerScriptEngineLegacy::callFunctionOnObjects(QList<QString> scriptFunctionPrefixes,
+        const QString& function,
+        QJSValueList args,
+        bool bFatalError) {
+    VERIFY_OR_DEBUG_ASSERT(m_pJSEngine) {
+        return false;
+    }
+
+    const QJSValue global = m_pJSEngine->globalObject();
+
+    bool success = true;
+    for (const QString& prefixName : scriptFunctionPrefixes) {
+        QJSValue prefix = global.property(prefixName);
+        if (!prefix.isObject()) {
+            qWarning() << "ControllerScriptHandlerLegacy: No" << prefixName << "object in script";
+            continue;
+        }
+
+        QJSValue init = prefix.property(function);
+        if (!init.isCallable()) {
+            qWarning() << "ControllerScriptHandlerLegacy:" << prefixName << "has no"
+                       << function << " method";
+            continue;
+        }
+        controllerDebug("ControllerScriptHandlerLegacy: Executing"
+                << prefixName << "." << function);
+        QJSValue result = init.callWithInstance(prefix, args);
+        if (result.isError()) {
+            showScriptExceptionDialog(result, bFatalError);
+            success = false;
+        }
+    }
+    return success;
+}
+
+QJSValue ControllerScriptEngineLegacy::wrapFunctionCode(
+        const QString& codeSnippet, int numberOfArgs) {
+    // This function is called from outside the controller engine, so we can't
+    // use VERIFY_OR_DEBUG_ASSERT here
+    if (m_pJSEngine == nullptr) {
+        return QJSValue();
+    }
+
+    QJSValue wrappedFunction;
+
+    auto i = m_scriptWrappedFunctionCache.constFind(codeSnippet);
```

Comment:
```suggestion
    const auto it = m_scriptWrappedFunctionCache.constFind(codeSnippet);
```

I sound like a number, but it's an iterator. And you shadow that variable in the for loop in line 65.